### PR TITLE
Replace AppVeyor with GitHub Actions

### PR DESCRIPTION
**Pull request by Or-Geva**
- [x] All [tests](https://ci.appveyor.com/project/jfrog-ecosystem/jenkins-artifactory-plugin) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is created in the [jfrog/jenkins-artifactory-plugin](https://github.com/jfrog/jenkins-artifactory-plugin) repository.
-----

### CODE REVIEW

**Review by yahavi — CHANGES_REQUESTED** [review]
I suggest also doing the following changes in the Gradle tests:

`gradleCiServerPublicationTest`: Keep the changes you made.

`gradleTest` and  `gradleCiServerTest`: Delete all changes, since we want to check old Gradle configurations. To make it works with Gradle 8 installed, let's use the Gradle wrapper. 
You can Use the wrapper by changing the `gradle.pipeline` and `gradleCiServer.pipeline`:
In scripted pipeline, add the following:
```groovy
rtGradle.useWrapper = true
```

In declarative pipeline, add the following in `rtGradleRun`:
```groovy
rtGradleRun (
  ...
  useWrapper: true
)

```

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - name: Activate Python virtual env on Posix
+        if: ${{ runner.os == 'macOS' || runner.os == 'Linux' }}
+        run: source venv/bin/activate
+
+      - name: Install Conan
+        run: pip install conan -q
+
+      - name: maven home on Linux
+        if: ${{ runner.os == 'Linux' }}
+        run: echo "MAVEN_HOME=$(whereis mvn)" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (Or-Geva) [reply .github/workflows/tests.yml:70]: Nope -> https://github.com/jfrog/jenkins-artifactory-plugin/blob/master/src/main/java/org/jfrog/hudson/maven3/Maven3Builder.java#L51

File: src/test/java/org/jfrog/hudson/pipeline/integration/CommonITestsPipeline.java [hunk 2]
```diff
@@ -366,7 +366,6 @@ void gradleTest(String buildName) throws Exception {
             assertEquals(4, buildInfo.getModules().size());
 
             assertModuleContainsArtifacts(buildInfo, "org.jfrog.example.gradle:services:1.0-SNAPSHOT");
-            assertModuleContainsArtifactsAndDependencies(buildInfo, "org.jfrog.example.gradle:api:1.0-SNAPSHOT");
```
Comment (yahavi) [inline src/test/java/org/jfrog/hudson/pipeline/integration/CommonITestsPipeline.java:369]: Please don't forget to revert this change
